dali/operators/sequence/element_extract.h

```diff
@@ -18,50 +18,87 @@
 
 #include <vector>
 #include "dali/core/common.h"
+#include "dali/core/format.h"
+#include "dali/core/tensor_layout.h"
 #include "dali/core/tensor_shape.h"
+#include "dali/kernels/common/scatter_gather.h"
 #include "dali/pipeline/operator/operator.h"
 
 namespace dali {
 
 namespace detail {
-  static void CheckInputShape(const TensorShape<>& tensor_shape,
+  static void CheckInputShape(const span<const int64_t>& tensor_shape,
                               const std::vector<int>& element_map) {
-    DALI_ENFORCE(tensor_shape.size() > 0);
+    assert(tensor_shape.size() > 0);
     auto N_input = tensor_shape[0];
 
-    int N_output = element_map.size();
-    DALI_ENFORCE(N_input >= N_output,
-        "Requested more elements than available");
-
     for (auto elem : element_map)
-        DALI_ENFORCE(elem < N_input,
-            "index " + std::to_string(elem) + " out of bounds");
+      DALI_ENFORCE(elem < N_input, "index " + std::to_string(elem) + " out of bounds");
```

Comment:
I would print the `N_input` value as well.

Reply:
Done